### PR TITLE
SDK: Silo Claim

### PR DESCRIPTION
**Pull request by 0xalecks**
## PR currently targets branch `silo-withdraw`. If that branch is merged first (#299 ), this PR needs to be updated to retarget master

Adds:
- sdk.silo.claim()
- sdk.silo.claimSeasons()
- sdk.silo.getClaimableAmount()

### CODE REVIEW

File: projects/sdk/src/lib/silo/Claim.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  /**
+   * Claims specific seasons from Silo claimable amount.
+   * @param token Which Silo token to withdraw. Must be a whitelisted token
+   * @param seasons Which seasons to claim, from the available claimable list. List of seasons
+   * can be retrieved with .getClaimableAmount()
+   * @param toMode Where to send the output tokens (circulating or farm balance)
+   */
+  async claimSeasons(token: Token, seasons: string[], toMode: FarmToMode = FarmToMode.EXTERNAL) {
```
Comment (silochad) [inline projects/sdk/src/lib/silo/Claim.ts:50]: I think we should have a higher-level discussion about naming variance between the SDK <> contract, but think this format (`claim` = claim all, `claimSeasons` = granular control) makes sense. Can discuss after Claim functionality is removed in zero withdraw BIP.

File: projects/sdk/src/lib/silo.ts [hunk 2]
```diff
@@ -91,6 +93,7 @@ export class Silo {
     Silo.sdk = sdk;
     this.depositBuilder = new DepositBuilder(sdk);
     this.withdraw = new Withdraw(sdk);
+    this.claim = new Claim(sdk);
```
Comment (silochad) [inline projects/sdk/src/lib/silo.ts:96]: Expanding on the `sdk.silo.claim.claim` note, feels strange to me to have `depositBuilder` (with "builder") but `withdraw` and `claim`. Understand though that the DepositBuilder is doing something functionally different, just thinking about this from an interface perspective.

**Review by silochad — APPROVED** [review]

File: projects/sdk/src/lib/silo/Claim.ts (hunk 1 above)
Comment (0xalecks) [reply projects/sdk/src/lib/silo/Claim.ts:50]: sounds good

File: projects/sdk/src/lib/silo.ts (hunk 2 above)
Comment (0xalecks) [reply projects/sdk/src/lib/silo.ts:96]: Yeah, it is weird, and it's something that bugged me too. The reason for depositBuilder, and swapBuilder, is to optimize in the UI. It gives you an object that has route calculated and allows you to play with the amounts w/o rebuilding the whole thing. We can discuss if there's way to improve this or if you think it's over-optimized.

**Pull request merged**